support-e2e/tests/genericCheckout.test.ts

```diff
@@ -0,0 +1,130 @@
+import { expect, test } from '@playwright/test';
+import { email, firstName, lastName } from './utils/users';
+import { setTestUserDetails } from './utils/testUserDetails';
+import { checkRecaptcha } from './utils/recaptcha';
+import { fillInCardDetails } from './utils/cardDetails';
+import { fillInDirectDebitDetails } from './utils/directDebitDetails';
+import { fillInPayPalDetails } from './utils/paypal';
+import { setupPage } from './utils/page';
+import { afterEachTasks } from './utils/afterEachTest';
+
+interface TestDetails {
+	product: 'Contribution' | 'SupporterPlus';
+	ratePlan: 'Monthly' | 'Annual';
+	paymentType?: 'Stripe' | 'DirectDebit' | 'PayPal';
+	price: number;
+	country?: 'UK' | 'US' | 'AU' | 'EU';
```

Comment:
You could make this explicit and add `UK` explicitly. Defaults create 1 extra step someone has to jump through to understand.

Reply:
Ok, this approach was taken in the `tieredCheckout.test.ts` but can see its unnecessary complication, `country` not optional now, done.